code/modules/food_and_drinks/food/snacks_pizza.dm

```diff
@@ -179,15 +179,15 @@
 	value = FOOD_ILLEGAL
 
 /obj/item/reagent_containers/food/snacks/proc/try_break_off(mob/living/M, mob/living/user) //maybe i give you a pizza maybe i break off your arm
+	if(prob(50) || (M != user) || !iscarbon(user) || HAS_TRAIT(user, TRAIT_NODISMEMBER))
```

Comment:
```suggestion
	if(M != user || !iscarbon(user) || HAS_TRAIT(user, TRAIT_NODISMEMBER) || prob(50))
```

Reply:
Is this a change you'd rather not have?

Reply:
Is it more performant to run all those checks before the prob check?

Reply:
Might be a micro-optimization, sure.